crd-generator/api-v2/src/main/java/io/fabric8/crdv2/generator/CRDGenerator.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crdv2.generator;
+
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import io.fabric8.crdv2.generator.v1.CustomResourceHandler;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.client.utils.ApiVersionUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinTask;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_EMPTY;
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+import static com.fasterxml.jackson.annotation.JsonInclude.Value.construct;
+
+public class CRDGenerator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CRDGenerator.class);
+  private final Map<String, AbstractCustomResourceHandler> handlers = new HashMap<>(2);
+  private CRDOutput<? extends OutputStream> output;
+  private boolean parallel;
+  private boolean implicitPreserveUnknownFields;
+  private ObjectMapper objectMapper;
+  private Map<String, CustomResourceInfo> infos;
+
+  // TODO: why not rely on the standard fabric8 yaml mapping
+  public static final ObjectMapper YAML_MAPPER = JsonMapper.builder(new YAMLFactory()
```

Comment:
Yes, why not?

Reply:
A comment was lost from @metacosm about how this format seemed different. Haven't looked at it more deeply.

Reply:
The two main changes I see are the standard generator includes the explicit start ---, and it quotes string values. The ordering changes aren't really needed as they should be handled by the construction of the crd object.

I agree that we don't need explicit start, the question is does it matter if there's additional quoting? It shouldn't but we can account for one or both of these differences with an KubernetesSerialization.asYaml method that takes some style options - we are just creating the Dump / DumpSettings on every call anyway.

Reply:
Looking at it more, I don't think we should add yaml options. They really aren't needed for this case either. We could just remove this yaml mapper - the only practical issue is that string comparison of the existing crd output yaml won't work.

Reply:
I can't find any reference going back in the code for that usage of a yaml mapper. The combination of the comment and the start marker are still legal yaml, so I'm not sure what the issue could would have been.

Reply:
Can confirm that the start marker is causing issues, in particular with OLM validation, at least.  I haven't tried applying the CRDs to a cluster, yet, though.

Reply:
Applying the CRD to the cluster seems to work OK, as expected.

Reply:
Thanks for checking @metacosm - the explicit start has been removed

Reply:
Stripping the marker solved the issue as expected. Thank you! 
For reference, here's the successful QOSDK build with the changes from this PR: https://github.com/quarkiverse/quarkus-operator-sdk/actions/runs/8903462629/job/24459014817

Reply:
Great, thank you for testing. The only other thing that could change is supplying the ObjectMapper / KuberenetesSerialization so that we can pick up any changes the user may have made to the mapper configuration

cc @manusa